agent/rpc_from.go

```diff
@@ -128,6 +128,25 @@ func (a *orbAgent) handleAgentStop(payload fleet.AgentStopRPCPayload) {
 
 func (a *orbAgent) handleAgentGroupRemoval(rpc fleet.GroupRemovedRPCPayload) {
 	a.unsubscribeGroupChannel(rpc.ChannelID)
+
+	policies, err := a.policyManager.GetRepo().GetAll()
+	if err != nil {
+		return
+	}
+
+	for _, policy := range policies {
+		delete(policy.GroupIds, rpc.AgentGroupID)
+
+		if len(policy.GroupIds) == 0 {
+			err = a.policyManager.RemovePolicy(policy.ID, policy.Name, policy.Backend)
+			if err != nil {
+				a.logger.Warn("failed to remove a policy, ignoring", zap.String("policy_id", policy.ID), zap.String("policy_name", policy.Name), zap.Error(err))
+				continue
+			}
+		}
+	}
+	delete(a.groupsInfos, rpc.AgentGroupID)
```

Comment:
i think this delete should happen in the `unsubscribeGroupChannel` method which is called above